src/private/CABasicAnimation+MotionAnimator.m

```diff
@@ -20,6 +20,22 @@
 
 #import <UIKit/UIKit.h>
 
+#pragma mark - Private
+
+static BOOL IsNumberValue(id someValue) {
+  return [someValue isKindOfClass:[NSNumber class]];
+}
+
+static BOOL IsValueType(id someValue, char *objCType) {
+  if ([someValue isKindOfClass:[NSValue class]]) {
+    NSValue *asValue = (NSValue *)someValue;
+    return strncmp(asValue.objCType, objCType, strlen(objCType)) == 0;
```

Comment:
If you're going to trust one of the strings, I would trust the `asValue.objCType` and not the `char *` passed in.

This whole thing seems safer as a list of functions:
```objectivec
static BOOL IsCGSizeValueType(id someValue);
static BOOL IsCGPointValueType(id someValue);
static BOOL IsCGRectValueType(id someValue);
```

Reply:
The thinking is that `id someValue` comes from user-land, while `objCType` comes from this file, so the vector of attack for `objCType` is smaller than that of `id someValue`.

Can you help me understand your thinking?

Reply:
Ah, I can see how you feel the specifically-typed methods would help make this safer, but I'm not certain that there's a large benefit when the APIs are private and scoped to this file. What do you think?

Reply:
I was thinking more of the "let's make this more general" risk later on. Someone copy-pastes this out into a header or utility class and keeps the behavior. In that case, the classes passed to this function are more likely to be first-party, but the `char *` could be anything.

Since within this class we only ever pass the result of the compiler `@encode` it's safe.  I don't think the really is a good "generic" as well as buffer-safe approach.  I'm OK merging this as-is since it is currently a static, internal function .

Reply:
Agreed with that thinking. I'll tidy this up so that copy-pasting is safer.